sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/JavaTypeInference.scala

```diff
@@ -166,317 +148,58 @@ object JavaTypeInference {
+      case pt: ParameterizedType => implementsInterface(pt.getRawType)
+      case c: Class[_] if interface.isAssignableFrom(c) => Option(c)
+      case _ => None
     }
 
-    if (!inputObject.dataType.isInstanceOf[ObjectType]) {
-      inputObject
-    } else {
-      typeToken.getRawType match {
-        case c if c == classOf[String] => createSerializerForString(inputObject)
-
-        case c if c == classOf[java.time.Instant] => createSerializerForJavaInstant(inputObject)
-
-        case c if c == classOf[java.sql.Timestamp] => createSerializerForSqlTimestamp(inputObject)
-
-        case c if c == classOf[java.time.LocalDateTime] =>
-          createSerializerForLocalDateTime(inputObject)
-
-        case c if c == classOf[java.time.LocalDate] => createSerializerForJavaLocalDate(inputObject)
-
-        case c if c == classOf[java.sql.Date] => createSerializerForSqlDate(inputObject)
-
-        case c if c == classOf[java.time.Duration] => createSerializerForJavaDuration(inputObject)
-
-        case c if c == classOf[java.time.Period] => createSerializerForJavaPeriod(inputObject)
-
-        case c if c == classOf[java.math.BigInteger] =>
-          createSerializerForBigInteger(inputObject)
-
-        case c if c == classOf[java.math.BigDecimal] =>
-          createSerializerForBigDecimal(inputObject)
-
-        case c if c == classOf[java.lang.Boolean] => createSerializerForBoolean(inputObject)
-        case c if c == classOf[java.lang.Byte] => createSerializerForByte(inputObject)
-        case c if c == classOf[java.lang.Short] => createSerializerForShort(inputObject)
-        case c if c == classOf[java.lang.Integer] => createSerializerForInteger(inputObject)
-        case c if c == classOf[java.lang.Long] => createSerializerForLong(inputObject)
-        case c if c == classOf[java.lang.Float] => createSerializerForFloat(inputObject)
-        case c if c == classOf[java.lang.Double] => createSerializerForDouble(inputObject)
-
-        case _ if typeToken.isArray =>
-          toCatalystArray(inputObject, typeToken.getComponentType)
-
-        case _ if ttIsAssignableFrom(listType, typeToken) =>
-          toCatalystArray(inputObject, elementType(typeToken))
-
-        case _ if ttIsAssignableFrom(mapType, typeToken) =>
-          val (keyType, valueType) = mapKeyValueType(typeToken)
-
-          createSerializerForMap(
-            inputObject,
-            MapElementInformation(
-              ObjectType(keyType.getRawType),
-              nullable = true,
-              serializerFor(_, keyType)),
-            MapElementInformation(
-              ObjectType(valueType.getRawType),
-              nullable = true,
-              serializerFor(_, valueType))
-          )
-
-        case other if other.isEnum =>
-          createSerializerForString(
-            Invoke(inputObject, "name", ObjectType(classOf[String]), returnNullable = false))
-
-        case other =>
-          val properties = getJavaBeanReadableAndWritableProperties(other)
```

Comment:
Hmm, I'm not sure if this has been discussed, but this is a breaking change causing customer issue like https://issues.apache.org/jira/browse/SPARK-48073.

Reply:
Is this an intentional change? @hvanhovell

Reply:
No matter what, this change has been there for a while. It seems difficult to change it back to old behavior. As discussed in the JIRA, I'd like to see if we can add a SQL conf to fall back to previous behavior of `Encoder.bean()`.

Reply:
The behavior was apparently there before I got to it, this is PR that fixed it later: https://github.com/apache/spark/pull/42829

For the record there is no specification here, only tests.

Reply:
Thanks. After more internal discussion with the customer, we decided not to make a change to previous behavior with a config.